### DISCUSSION TITLE
[Question] Center data in a DataTable

### DISCUSSION THREAD

**Discussion by ingdesarr1** [flet-dev/flet#2488] [Q&A] [answered]
### Question

Good afternoon. Is there a way to center the content of the cells for the datatable? When DataCells are added to DataRows, the content is aligned to the left and it would be better to be able to center it. Thank you

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by wschume** [comment]
As far as I can tell, the flutter [DataTable](https://api.flutter.dev/flutter/material/DataTable-class.html) does not come with alignment. Hence, it is not possible to align the text in a Flet DataTable as well.

**Comment by burhansvural** [comment, marked as answer]
@ingdesarr1  You can wrap the control you want to add to the DataCell with a Container and use its alignment property.

![datatable_centered_data](https://github.com/flet-dev/flet/assets/149883930/d4e5db5f-1e94-4572-8e46-e6a30f9f1b93)

`import flet as ft

def main(page: ft.Page):
    page.add(
        ft.DataTable(
            columns=[
                ft.DataColumn(ft.Text("First name")),
                ft.DataColumn(ft.Text("Last name")),
                ft.DataColumn(ft.Text("Age"), numeric=True),
            ],
            rows=[
                ft.DataRow(
                    cells=[
                        ft.DataCell(
                            content=ft.Container(
                                alignment=ft.alignment.center,
                                content=ft.Text("John")
                            )
                            ),
                        ft.DataCell(
                            content=ft.Container(
                                alignment=ft.alignment.center,
                                content=ft.Text("Smith")
                                )),
                        ft.DataCell(ft.Text("43")),
                    ],
                ),
                ft.DataRow(
                    cells=[
                        ft.DataCell(
                            content=ft.Container(
                                alignment=ft.alignment.center,
                                content=ft.Text("Jack")
                                ),
                            ),
                        ft.DataCell(
                            content=ft.Container(
                                alignment=ft.alignment.center,
                                content=ft.Text("Brown")
                                ),
                            ),
                        ft.DataCell(ft.Text("19")),
                    ],
                ),
                ft.DataRow(
                    cells=[
                        ft.DataCell(
                            content=ft.Container(
                                alignment=ft.alignment.center,
                                content=ft.Text("Alice")
                                ),
                            ),
                        ft.DataCell(
                            content=ft.Container(
                                alignment=ft.alignment.center,
                                content=ft.Text("Wong")
                                ),
                            ),
                        ft.DataCell(ft.Text("25")),
                    ],
                ),
            ],
        ),
    )

ft.app(target=main)`
